File: Chapter11/Java/GraphNode.java
package main.java;

import java.util.LinkedList;

/**
 * Created by William Smith on 10/14/16.
 */
public class GraphNode
{
    public int Value;

    private LinkedList<GraphNode> _neighbors;
    public LinkedList<GraphNode> GetNeighbors()
    {
        return _neighbors;
    }

    public GraphNode()
    {
        _neighbors = new LinkedList<GraphNode>();
    }

    public GraphNode(int value)
    {
        _neighbors = new LinkedList<GraphNode>();
        Value = value;
    }
}

File: Chapter11/Swift/GraphNode.swift
//
//  GraphNode.swift
//  EverydayDataStructures-Swift
//
//  Created by William Smith on 10/14/16.
//  Copyright © 2016 Websmiths, LLC. All rights reserved.
//

import Foundation

public class GraphNode : Equatable
{
    public var neighbors: Array = [GraphNode]()
    public var value : Int
    
    public init(val: Int) {
        value = val
    }
}

public func == (lhs: GraphNode, rhs: GraphNode) -> Bool {
    return (lhs.value == rhs.value)
}
